pkg/providers/aws/provider_postgres.go

```diff
@@ -444,8 +455,44 @@ func (p *PostgresProvider) deleteRDSInstance(ctx context.Context, pg *v1alpha1.P
 	}
 
 	// check if instance does not exist, delete finalizer and credential secret
-	if foundInstance == nil {
-		//TODO will be implemented and tested correctly in - https://issues.redhat.com/browse/INTLY-8103
+	if foundInstance != nil {
+		// set status metric
+		p.exposePostgresMetrics(ctx, pg, foundInstance)
+
+		// return if rds instance is not available
+		if *foundInstance.DBInstanceStatus != "available" {
+			statusMessage := fmt.Sprintf("delete detected, deleteDBInstance() in progress, current aws rds status is %s", *foundInstance.DBInstanceStatus)
+			logger.Info(statusMessage)
+			return croType.StatusMessage(statusMessage), nil
+		}
+
+		// delete rds instance if deletion protection is false
+		if !*foundInstance.DeletionProtection {
+			_, err = instanceSvc.DeleteDBInstance(rdsDeleteConfig)
+			rdsErr, isAwsErr := err.(awserr.Error)
+			if err != nil && (!isAwsErr || rdsErr.Code() != rds.ErrCodeDBInstanceNotFoundFault) {
+				msg := fmt.Sprintf("failed to delete rds instance : %s", err)
+				return croType.StatusMessage(msg), errorUtil.Wrapf(err, msg)
+			}
+			return "delete detected, deleteDBInstance() started", nil
+		}
+
+		// modify rds instance to turn off deletion protection
+		_, err = instanceSvc.ModifyDBInstance(&rds.ModifyDBInstanceInput{
+			DBInstanceIdentifier: rdsDeleteConfig.DBInstanceIdentifier,
+			DeletionProtection:   aws.Bool(false),
+		})
+		if err != nil {
+			msg := "failed to remove deletion protection"
+			return croType.StatusMessage(msg), errorUtil.Wrap(err, msg)
+		}
+
+		return croType.StatusMessage(fmt.Sprintf("deletion protection detected, modifyDBInstance() in progress, current aws rds status is %s", *foundInstance.DBInstanceStatus)), nil
+	}
+
+	// standaloneNetworkExists if no bundled resources are found in the cluster vpc
+	if standaloneNetworkExists && isLastResource {
```

Comment:
Just a concern, when we `DeleteNetworkConnection` we remove the route table entry in the cluster route table. A valid thing to do IMO, but we require the peering connection to do this. The issue is, if an error happens when `DeleteNetwork` is called, we re-reconcile, we `GetClusterNetworkPeering`, then try `DeleteNetworkPeering` this will return, as the peering connection does not exist, and the `networkPeering` object will be nil, we call `DeleteNetworkConnection` with a nil object we return an error. We never get passed this error and the original error is lost. 

We should look to update the logic in `DeleteNetworkConnection` to handle this flow better. As the route table entry is a nice to have, we can probably do one of a few things. 

1. If the `networkPeering` connection is nil, we can return and not error or try remove the route table entries. 
2. When the peering connection begins to delete the route table entry goes into `blackhole` status, we can check for `blackhole` status and the standalone vpc cidr block. And delete the route table entry based on those fields existing. 
3. Maybe a discussion around on how necessary removing these entries are from the cluster VPC, chances are if RHMI is being deleted the cluster will be nukes and the route table will be deleted with the cluster VPC

What do you think, might be worth a call/discussion about this.

Reply:
What's the issue with just deleting the route table entry before the peering connection?

Reply:
no issue, that would probably be the easiest fix, to alternate the flow of the order we remove items. Remove the connection -> remote the peering -> remove the network.

Reply:
Let's go with that then, unless you see any issues @laurafitzgerald ?

Reply:
@aidenkeating @ciaranRoche the flow is now `Remove the connection -> remote the peering -> remove the network.` in case postgres and redis. 👍